lib/index.js

```diff
@@ -263,42 +312,144 @@ api.get = async ({actor, account, email, type} = {}) => {
   } else {
     query['account.email'] = email;
   }
+
+  const tokenId = 'meta.bedrock-authn-token.tokens.nonce.id';
+  if(id) {
+    query[tokenId] = id;
+  }
+
   const record = await database.collections.account.findOne(
     query, {projection: {_id: 0, [typeKey]: 1}});
+
+  if(!record) {
+    throw new BedrockError(
+      'Authentication token not found.',
+      'NotFoundError', {
+        httpStatusCode: 404,
+        public: true
+      });
+  }
+
+  let result = record.meta['bedrock-authn-token'].tokens[type];
+  let rValue;
+  if(type === 'nonce') {
+    if(id) {
+      result = result.filter(token => token.id === id);
+    }
+    rValue = result[0];
+  } else {
+    rValue = result;
+  }
+  // filter expired token.
+  if(filterExpiredTokens) {
+    const now = Date.now();
+    if(rValue.expires < now) {
+      // throw NotFoundError
+      throw new BedrockError(
+        'Authentication token has expired.',
+        'NotFoundError', {
+          httpStatusCode: 404,
+          public: true
+        });
+    }
+  }
+  return rValue;
+};
+
+/**
+ * Gets tokens for an account.
+ *
+ * @param {object} options - The options to use.
+ * @param {object} options.actor - The actor or capabilities to perform
+ *   the action.
+ * @param {string} [options.account] - The ID of the account.
+ * @param {string} [options.email] - The email of the account.
+ * @param {string} options.type - The type of tokens to get.
+ * @param {boolean} [options.filteredExpiredTokens] - Set to `true` to drop
+ *   expired tokens and `false` to get all tokens including expired tokens.
+ *
+ * @return {Promise<array>} - A Promise that resolves to tokens.
+ */
+api.getAll = async ({
+  actor, account, email, type, filterExpiredTokens = true
+} = {}) => {
+  assert.optionalString(account, 'account');
+  assert.optionalString(email, 'email');
+  assert.optionalBool(filterExpiredTokens, 'filterExpiredTokens');
+
+  if(!(account || email)) {
+    throw new Error('Either "account" or "email" must be provided.');
+  }
+  validateTokenType(type);
+
+  await brPermissionCheck(
+    actor, PERMISSIONS.ACCOUNT_ACCESS, {resource: [account]});
+
+  // get tokens from meta
+  const typeKey = `meta.bedrock-authn-token.tokens.${type}`;
+  const query = {[typeKey]: {$exists: true}};
+  if(account) {
+    query.id = database.hash(account);
+  } else {
+    query['account.email'] = email;
+  }
+  const record = await database.collections.account.findOne(
+    query, {projection: {_id: 0, [typeKey]: 1}});
+
   if(!record) {
```

Comment:
Minor edge case here, unsure if it matters in practice.  If you had never set a nonce, you would get no record and throw a 404.  But if you ever had set nonces and removed them all leaving an empty array, or if you have nonces but they are all expired, then it will fall through and return an empty array.

Reply:
We should unify -- and I'd think we could do that by promoting the `expires` filter check into the query, right?

Reply:
I'm not sure what you're suggesting.

Reply:
I was suggesting that we could include the `expires` check that is done below in a filter function in the mongo query using `$gt` or whatever is appropriate. However, upon reflection, since this is a `getAll` API, we should only throw a not found if *the account* doesn't exist ... and always return an empty array when the account exists but there are no (unexpired) tokens that match the query.

Reply:
Updated.